go/libraries/doltcore/servercfg/serverconfig.go

```diff
@@ -125,8 +125,13 @@ type ServerConfig interface {
 	Port() int
 	// User returns the username that connecting clients must use.
 	User() string
+	// UserIsSpecified returns true if a user was explicitly provided in the configuration.
+	UserIsSpecified() bool
 	// Password returns the password that connecting clients must use.
 	Password() string
+	// SkipRootUserInitialization returns whether the server should skip initializing the default root superuser.
+	// This option is only valid the first time a sql-server is started, when the root user is being initialized.
+	SkipRootUserInitialization() bool
```

Comment:
Would be better to keep this out of the interface if possible, see comments below